conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -363,20 +379,20 @@ def content(self):
         host_req = self._conanfile.dependencies.host
         build_req = self._conanfile.dependencies.build  # tool_requires
         test_req = self._conanfile.dependencies.test
-
-        # Check if it exists both as require and as build require without a suffix
-        self._validate_build_requires(host_req, build_req)
+        # If self.build_context_suffix is not defined, the build requires will be saved
+        # in the self.build_context_folder
+        if self.build_context_suffix:
+            self._conanfile.output.warning("build_context_suffix attribute has been deprecated. Use the"
+                                           " build_context_folder one instead.")
+            # Check if it exists both as require and as build require without a suffix
+            self._validate_build_requires(host_req, build_req)
```

Comment:
Condition the validate to the definition of ``build_context_folder``. If it is not defined, it should validate, as it was doing before.

Reply:
My fault, I thought that I've already pushed those changes